solr/test-framework/src/java/org/apache/solr/util/SolrClientTestRule.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.util;
+
+import static org.apache.solr.EmbeddedSolrServerTestBase.solrClientTestRule;
+
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.nio.file.Path;
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.request.UpdateRequest;
+import org.junit.rules.ExternalResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * SolrClientTestRule is an abstract class that extends JUnit's {@link ExternalResource} class,
+ * which is a TestRule that allows for specifying methods to be run before and after a test.
+ * SolrClientTestRule provides access to a {@link SolrClient} instance. SolrClientTestRule should be
+ * used in tests that need to interact with a Solr server and provides a convenient way to manage
+ * the lifecycle of the SolrClient instance.
+ */
+public abstract class SolrClientTestRule extends ExternalResource {
+
+  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+  protected SolrClientTestRule() {}
+
+  @Override
+  protected void before() throws Throwable {
+    super.before();
+  }
+
+  public abstract SolrClient getSolrClient();
+
+  public void clearIndex() throws SolrServerException, IOException {
+    new UpdateRequest().deleteByQuery("*:*").commit(solrClientTestRule.getSolrClient(), null);
+  }
+
+  public abstract Path getSolrHome();
```

Comment:
needs docs.
BTW I recall there is some doubt as to wether we actually need this.  I suspect we don't?  If we can eliminate it -- even better.